### DISCUSSION TITLE
colmap CMAKE_CUDA_DEVICE_LINK_EXECUTABLE error

### DISCUSSION THREAD

**Discussion by supdhn** [colmap/colmap#1575] [Q&A]
Hi Guys

I am following the step by step guide from (https://colmap.github.io/install.html#pre-built-binaries) but I am getting the following error after the "cmake .." command:
_CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_CUDA_DEVICE_LINK_EXECUTABLE_

and when executing the make -j 8 I get:
_[100%] Linking CXX executable colmap
c++: error: CMakeFiles/colmap_exe.dir/cmake_device_link: No such file or directory
src/exe/CMakeFiles/colmap_exe.dir/build.make:425: recipe for target 'src/exe/colmap' failed
make[2]: *** [src/exe/colmap] Error 1
CMakeFiles/Makefile2:1105: recipe for target 'src/exe/CMakeFiles/colmap_exe.dir/all' failed
make[1]: *** [src/exe/CMakeFiles/colmap_exe.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2_

I am on an Ubuntu 18. 04 and I have installed Nvidia driver 515.57 and cuda 11.7

What I am missing ? what do I have to do to set this variable ?

Any help is appreciated.

tks in advance

**Comment by Zmy6** [comment]
I got the same problem!
Have you solved it or anyone can solve this issue?
Thanks!

**Reply by ihasime** [reply]
Hi,

sorry for the late reply. Unfortunately I was not able to find a solution... yet. Since I am on a tight schedule, I switched to windows and I am using the pre-built binaries but I will get back to this as soon as I can. 

best regards

**Reply by shubham-monarch** [reply]
Did you try `cmake .. -GNinja -DCMAKE_CUDA_ARCHITECTURES=all`?
